src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+            gSaveContext.infTable[15] |= 0x10;
+        }
+        return 0;
+    } else {
+        return 1;
+    }
+}
+
+u16 EnGo2_getTextIdGoronCityLowestFloor(GlobalContext* globalCtx, EnGo2* this) {
+    if (CHECK_QUEST_ITEM(QUEST_MEDALLION_FIRE) && LINK_IS_ADULT) {
+        return 0x3043;
+    } else if (CHECK_QUEST_ITEM(QUEST_GORON_RUBY)) {
+        return 0x3027;
+    } else {
+        return CUR_UPG_VALUE(UPG_STRENGTH)
+                   ? 0x302C
+                   : Flags_GetSwitch(globalCtx, 0x1B) == 0 ? 0x3017
+                                                           : gSaveContext.infTable[15] & 0x100 ? 0x3019 : 0x3018;
+    }
+}
+
+u16 EnGo2_getStateGoronCityLowestFloor(GlobalContext* globalCtx, EnGo2* this) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 2) {
+        if (this->actor.textId == 0x3018) {
+            gSaveContext.infTable[15] |= 0x100;
+        }
+        return 0;
+    } else {
+        return 1;
+    }
+}
+
+u16 EnGo2_getTextIdGoronCityLink(GlobalContext* globalCtx, EnGo2* this) {
+    if (CHECK_QUEST_ITEM(QUEST_MEDALLION_FIRE)) {
+        return gSaveContext.infTable[16] & 0x8000 ? 0x3042 : 0x3041;
+    } else if (CHECK_OWNED_EQUIP(EQUIP_TUNIC, 1)) {
+        return gSaveContext.infTable[16] & 0x4000 ? 0x3038 : 0x3037;
+    } else if (gSaveContext.infTable[16] & 0x1000) {
+        this->unk_20C = 0;
+        this->dialogState = 0;
+        return gSaveContext.infTable[16] & 0x400 ? 0x3033 : 0x3032;
+    } else {
+        return 0x3030;
+    }
+}
+
+u16 EnGo2_getStateGoronCityLink(GlobalContext* globalCtx, EnGo2* this) {
+    switch (EnGo2_GetDialogState(this, globalCtx)) {
+        case 2:
+            switch (this->actor.textId) {
+                case 0x3036:
+                    EnGo2_GetItem(this, globalCtx, GI_TUNIC_GORON);
+                    this->actionFunc = EnGo2_InitGetItem;
+                    return 2;
+                case 0x3037:
+                    gSaveContext.infTable[16] |= 0x4000;
+                default:
+                    return 0;
+            }
+        case 4:
+            if (func_80106BC8(globalCtx)) {
+                if (this->actor.textId == 0x3034) {
+                    if (globalCtx->msgCtx.choiceIndex == 0) {
+                        this->actor.textId = gSaveContext.infTable[16] & 0x800 ? 0x3033 : 0x3035;
+                        if (this->actor.textId == 0x3035) {
+                            func_800F8D04(0x39EB);
+                        }
+                    } else {
+                        this->actor.textId = gSaveContext.infTable[16] & 0x800 ? 0x3036 : 0x3033;
+                        if (this->actor.textId == 0x3036) {
+                            func_800F8D04(0x39EB);
+                        }
+                    }
+                    func_8010B720(globalCtx, this->actor.textId);
+                    this->unk_20C = 0;
+                }
+            } else {
+                break;
+            }
+            return 1;
```

Comment:
This `break;` jumps to the `return 1;` at the end of the function, so maybe you can get away with replacing these four lines with just `break;` or `return 1;`? e.g.
```suggestion
            break;
```

Might not work but worth trying. There are similar instances of this in the next function too

Reply:
I see your point in that it's redundant, but it doesn't work unfortunately. Same in the next function too. I'm not sure why, it would make sense to match with that suggestion.